src/java.desktop/windows/data/fontconfig/fontconfig.properties

```diff
@@ -198,59 +199,59 @@ dialoginput.bolditalic.korean=Gulim
-
-sequence.serif.GB18030=alphabetic,chinese-gb18030,dingbats,symbol,chinese-gb18030-extb
-sequence.sansserif.GB18030=alphabetic,chinese-gb18030,dingbats,symbol,chinese-gb18030-extb
-sequence.monospaced.GB18030=chinese-gb18030,alphabetic,dingbats,symbol,chinese-gb18030-extb
-sequence.dialog.GB18030=alphabetic,chinese-gb18030,dingbats,symbol,chinese-gb18030-extb
-sequence.dialoginput.GB18030=alphabetic,chinese-gb18030,dingbats,symbol,chinese-gb18030-extb
-
-sequence.serif.x-windows-950=alphabetic,chinese-ms950,dingbats,symbol,chinese-ms950-extb
-sequence.sansserif.x-windows-950=alphabetic,chinese-ms950,dingbats,symbol,chinese-ms950-extb
-sequence.monospaced.x-windows-950=chinese-ms950,alphabetic,dingbats,symbol,chinese-ms950-extb
-sequence.dialog.x-windows-950=alphabetic,chinese-ms950,dingbats,symbol,chinese-ms950-extb
-sequence.dialoginput.x-windows-950=alphabetic,chinese-ms950,dingbats,symbol,chinese-ms950-extb
-
-sequence.serif.x-MS950-HKSCS=alphabetic,chinese-ms950,chinese-hkscs,dingbats,symbol,chinese-ms950-extb
-sequence.sansserif.x-MS950-HKSCS=alphabetic,chinese-ms950,chinese-hkscs,dingbats,symbol,chinese-ms950-extb
-sequence.monospaced.x-MS950-HKSCS=chinese-ms950,alphabetic,chinese-hkscs,dingbats,symbol,chinese-ms950-extb
-sequence.dialog.x-MS950-HKSCS=alphabetic,chinese-ms950,chinese-hkscs,dingbats,symbol,chinese-ms950-extb
-sequence.dialoginput.x-MS950-HKSCS=alphabetic,chinese-ms950,chinese-hkscs,dingbats,symbol,chinese-ms950-extb
-
-sequence.serif.x-MS950-HKSCS-XP=alphabetic,chinese-ms950,chinese-hkscs,dingbats,symbol,chinese-ms950-extb
-sequence.sansserif.x-MS950-HKSCS-XP=alphabetic,chinese-ms950,chinese-hkscs,dingbats,symbol,chinese-ms950-extb
-sequence.monospaced.x-MS950-HKSCS-XP=chinese-ms950,alphabetic,chinese-hkscs,dingbats,symbol,chinese-ms950-extb
-sequence.dialog.x-MS950-HKSCS-XP=alphabetic,chinese-ms950,chinese-hkscs,dingbats,symbol,chinese-ms950-extb
-sequence.dialoginput.x-MS950-HKSCS-XP=alphabetic,chinese-ms950,chinese-hkscs,dingbats,symbol,chinese-ms950-extb
-
-sequence.allfonts.UTF-8.hi=alphabetic/1252,devanagari,dingbats,symbol
-sequence.allfonts.UTF-8.ja=alphabetic,japanese,dingbats,symbol
-sequence.allfonts.UTF-8.ko=alphabetic,korean,dingbats,symbol
-
-sequence.allfonts.windows-1255=hebrew,alphabetic/1252,dingbats,symbol
-
-sequence.serif.windows-31j=alphabetic,japanese,dingbats,symbol
-sequence.sansserif.windows-31j=alphabetic,japanese,dingbats,symbol
-sequence.monospaced.windows-31j=japanese,alphabetic,dingbats,symbol
-sequence.dialog.windows-31j=alphabetic,japanese,dingbats,symbol
-sequence.dialoginput.windows-31j=alphabetic,japanese,dingbats,symbol
-
-sequence.serif.x-windows-949=alphabetic,korean,dingbats,symbol
-sequence.sansserif.x-windows-949=alphabetic,korean,dingbats,symbol
-sequence.monospaced.x-windows-949=korean,alphabetic,dingbats,symbol
-sequence.dialog.x-windows-949=alphabetic,korean,dingbats,symbol
-sequence.dialoginput.x-windows-949=alphabetic,korean,dingbats,symbol
```

Comment:
I'd like to understand the reasoning behind the changes I see in the windows fontconfig.properties

The effect of what I see is (if I have it right)
1) Make Segoe UI Symbol a "core" part of every font - with the consequence that you potentially change
the metrics of every logical font- something which is often a compatibility issue
2) You then go ahead and exclude virtually every code point in that font, which kind of
defeats the point of having it at all .. and is  I guess you want it to take precedence for
all those code points over the emoji font .. but not some others ??
It seems like additions in later versions of either font would mean there'd be updates
needed here.
And FWIW if the Emoji font isn't available they will now get missing glyphs

also you are making a policy decision here that apps which currently get ordinary
symbol chars will now get emojis for those codepoints, whether they want them or not.

I think need to look at what code points these fonts support ..  IMO the Emoji font
as a fall back should be for unicode specified Emoji code points which I expect
not to see in the normal symbol font ..

In other words, what I would *expect* is that you just added the emoji as another
fallback (somewhere after symbols) and changed NOTHING else in this file.

Reply:
Oh if only I remembered the details. Overall, such complex changes were made to support variation selectors. See: we have Segoe UI Emoji for colored emoji and Segoe UI Symbol for monochrome ones, each emoji by default must be rendered either colored or monochrome according to CLDR, so I add Segoe UI Symbol into fallback sequence, but exclude those ranges which shouldn't be rendered monochrome by default (that is, most of the characters). Then to make variation selector work, we ignore these exclusion ranges if monochrome selector is used or something like that. So this is some kind of hack to make variation selectors work without "branching" the fallback sequence and fit into existing architecture. I will look through the code to refresh the details and will let you know more.

Reply:
Please do provide details .. but I don't think we need to .. or should .. be making variation selectors work where the variations aren't supported by a single physical font.

Reply:
Why shouldn't we? That's how it works on Windows - monochrome emoji in Segoe UI Symbos and colored in Segoe UI Emoji, not very convenient, but I believe we have to deal with it. Native apps do support this and I guess OS font rendering has some emoji-specific branching to support this.

Reply:
Are you sure ? I would expect that Segoe UI Emoji has a glyf table as well as a colr table.

Reply:
I thought so, but now you made me doubt. Anyway, I will read the code, recall this and will let you know what I find.

Reply:
Well, what I did find so far:
* Segoe UI Emoji does have `FE0F` UVS (colored selector), but no `FE0E` (monochrome).
* Segoe UI Symbol doesn't have any UVS.

So fonts alone will not guide us to a correct glyph.
However, we can ignore COLR table and render base glyphs from Segoe UI Emoji to achieve monochrome effect, so we actually can have both mono/color inside Segoe UI Emoji alone (but we would still have to determine whether to use mono or color ourselves).
But these base glyphs are not the same as in Segoe UI Symbol and I can clearly see that every Windows app which respect variation selectors, renders monochrome ones using Segoe UI Symbol, not base glyphs from Segoe UI Emoji.

I will try to investigate this further, but so far it looks to me like to support variation selectors the way native apps do, we need to use some kind of hardcoded tables, like those exclusion ranges, because:
1. We don't know whether character must be draw mono/colored by default, fonts doesn't tell us, only CLDR does.
2. Even if we somehow know the default variant, looking for `FE0F` selector will lead us to to Segoe UI Emoji, but `FE0E` - nowhere.

Ideas?

Reply:
So I restored the original idea.
1. Place Symbol into main font sequences to preceed the Emoji and be able to have exclusion ranges
2. Exclude everything that should be colored by default
3. One more hack in composite mapper - ignore exclude ranges when variation selector is specified

Results:
* No selector, default mono: will be handled by Symbol
* No selector, default color: skip Symbol because excluded, handle by Emoji
* FE0F (color): skip Symbol because it doesn't define mapping for this selector, handle by Emoji
* FE0E (mono): skip Symbol & Emoji & remaining fonts because they don't define this selector. No glyph found, so try again without selector, but ignoring exclusion ranges - handle by Symbol

Reply:
The best idea that came to my mind so far is to insert proxy "emoji" mapper into fallback sequence, which will handle default variations and stuff and then forward calls to real mono/color mapper. And this logic can also be reused in other OSes. What do you think?

Reply:
If the problem we are trying to solve here is around which code points have a default emoji presentation then I think we can have an internal table based on Unicode 15's emoji-data.txt
The JDK sources include a copy of that in java.base, but I don't propose we access it - just use it to hand build a table of the 1205 default emoji.

If that ISN'T the problem you are trying to solve then you'll need to explain it again.
Also I had a VERY brief exchange about whether java.lang.Character could expose this info to us - built directly from that CLDR data. That may be the longer term solution which might start out as JDK internal API.

Then how to fiddle around multiple fonts to do this is the next question.
I really don't want to have knowledge of the code points a particular font supports
(ie the exclusion ranges) but if we just do a lookup on that table and its a hit, then we can go looking through any fonts with a  COLR table for an emoji glyph. Otherwise no need to do anything. 
We'd need to convince ourselves that it is OK that inserting a VS for emoji=off could make us find a different design from a different font. 
Or we just put the emoji font before the symbol font .. 

And as for what "native apps do", tell me how you tested this.
In my testing on Windows 11 "WordPad aka Write" seems incapable of displaying color emoji.
Notepad can .. but the variation selector is not doing anything.
And I'm not sure what font glyphs come from but I don't see it mattering as long as we do something like the above

Reply:
Yes, that's the problem I'm trying to solve - determine default mono/emoji presentation for a code point.
And second part of this problem is how to alter this presentation with variation selectors.
I will try this approach with the lookup table and then we'll discuss this in more details.

Regarding native apps, I tested this on Windows 11 (and Windows 10 when I was implementing this) by copying this string into textfields of different apps:
❤ ❤︎ ❤️ 🔥 🔥︎ 🔥️
It's a plain emoji, then with mono selector, then with emoji selector, then repeat with different unicode.
The difference is that "heart" emoji is mono and "fire" is colored by default, so if app correctly handles VS, it should display like this:
M M C C M C

My results:
Notepad - works correctly, including VS. Mono from Symbol and colored from Emoji.
WordPad - uses Symbol (mono) with plain unicode, base (mono) glyphs from Emoji font when VS is present.
Address bar in Explorer - always mono, using base glyphs from Emoji font.
Google Chrome - works correctly, like Notepad if using search field on google.com, always colored when using address bar, preview of this comment shows all colored as well.